### PR TITLE
Set use_reentrant to False for gradient checkpointing

### PR DESCRIPTION
**Pull request by arnavgarg1**
Fixes the following warnings in newer versions of Torch (>= 2.1.0)

```
/home/ray/anaconda3/lib/python3.8/site-packages/torch/utils/checkpoint.py:429: UserWarning: torch.utils.checkpoint:
 please pass in use_reentrant=True or use_reentrant=False explicitly. The default value of use_reentrant will be 
updated to be False in the future. To maintain current behavior, pass use_reentrant=True. It is recommended that 
you use use_reentrant=False. Refer to docs for more details on the differences between the two variants.
```

and 

```
`use_cache=True` is incompatible with gradient checkpointing. Setting `use_cache=False`...
```

### CODE REVIEW

**Review by alexsherstinsky — APPROVED** [review]
LGTM!  Thank you so much for the solution for this ever-present warning!
